cclib/parser/gaussianparser.py

```diff
@@ -184,6 +184,23 @@ def after_parsing(self):
             for cpu_time in self.metadata["cpu_time"]:
                 self.metadata["wall_time"].append(cpu_time / self.num_cpu)
 
+        # Some calculation types don't use consistent MP levels throughout.
+        # For example, EOM-CCSD + opt prints MP3 and MP4 energies for the first iteration only.
+        # However, mpenergies needs to be homogeneous, so we need to do something about it.
+        if hasattr(self, "mpenergies"):
+            max_mp = max(len(mp_e) for mp_e in self.mpenergies)
+            mp_energies = []
+            for index, energy in enumerate(self.mpenergies):
+                if len(energy) != max_mp:
+                    self.logger.warning(
+                        f"MP energies of order {index+2} are incomplete and will be ignored"
```

Comment:
Huh I never knew the logging calls supported substitutions directly, how fun.

The use of the format string was introduced by the linter FYI (I still like the old format() method), hopefully it has a setting to prefer this style. Is it ruff we're using?

Reply:
Yes. I may not have turned on the correct ruff rule, since pylint is able to catch it. AFAIK the logging module is the only location this is true: https://medium.com/swlh/why-it-matters-how-you-log-in-python-1a1085851205